### PR TITLE
Fix recursive directory navigation

### PR DESCRIPTION
**Pull request by saurabh3949**
Since event.name of inotify just returns the folder name if a new folder is created, the directory structure wasn't being walked properly. This change fixes the bug to enable correct traversal of the directory.

### CODE REVIEW

File: src/sagemaker_containers/_intermediate_output.py [hunk 1]
```diff
@@ -97,7 +97,9 @@ def _watch(inotify, watchers, watch_flags, s3_uploader):
                 # for it which should cause any problems because when we copy files to temp dir
                 # we add a unique timestamp up to microseconds.
                 if flag is inotify_simple.flags.ISDIR and inotify_simple.flags.CREATE & event.mask:
-                    for folder, dirs, files in os.walk(os.path.join(intermediate_path, event.name)):
+                    path = os.path.join(
+                    *[intermediate_path] + [watchers[wd] for wd in range(1, event.wd + 1)] + [event.name])
```
Comment (mvsusp) [inline src/sagemaker_containers/_intermediate_output.py]: Please extract this logic outside `os.path.join` to be more readable and easier to understand.

Comment (saurabh3949) [reply src/sagemaker_containers/_intermediate_output.py]: There was an issue with this. I have fixed it in the latest commit.

**Comment by nadiaya** [comment]
We have a test that was supposed to test for this behavior, but failed :(
We should add test that creates dir1 and dir1/dir2 with a sleep in between them.

**Comment by nadiaya** [comment]
Tests failed because of this numpy bug:
https://github.com/numpy/numpy/pull/12754

**Pull request merged**